sockeye/training.py

```diff
@@ -401,30 +427,29 @@ def _evaluate(self, checkpoint: int, data_iter,
         for batch in data_iter:
             batch = batch.load(device=self.device)
             with torch.inference_mode():
-                # Forward: use sockeye_model because (traced) training_model
```

Comment:
that comment might still be valuable to have. A user might wonder why not use `model_object.model` here instead.

More generally: would it make sense to wrap all of the different model instances and losses into one object, i.e. 
```
ModelWithLoss, containing
- model (traced)
- model (untraced)
- losses
```

Reply:
This gets more complicated when we add DeepSpeed support, which automatically handles saving/loading the model object's state. If we have a regular model and a traced model, it tries to load parameters for both. The traced parameters don't exist, so we need additional logic to filter them out. With a single model instance, the `ModelWithLoss` object state can saved/loaded as a standard Pytorch module.

This also helps us minimize complexity. We have a `ModelWithLoss` object that turns input batches into loss outputs using the fully traced/wrapped model. We also have the "vanilla" `SockeyeModel` for saving parameters, running the checkpoint decoder, etc. This way the object with added complexity (`ModelWithLoss`) only needs to support a single use case.

I've added back a version of the comment about why we use the `SockeyeModel` here.